src/Misc/MSet.jl

```diff
@@ -44,70 +191,372 @@ function Base.show(io::IO, ::MIME"text/plain", s::MSet)
         print(io, "$k")
       end
     end
-    print(io,")")
+    print(io, ")")
+  end
 end
 
+# We use the one-line (hopefully implemented!) printing system for the elements
+# in
+function Base.show(io::IO, ::MIME"text/plain", s::MSet)
+  print(io,"MSet{",eltype(s),"}")
+  if isempty(s)
+    print(io,"()")
+  else
+    io = pretty(io)
+    szh, szw = displaysize(io)
+    szh -= 5
+    szw -= 10
+    print(io, " with ", ItemQuantity(length(s), "element"), ":")
+    print(io, Indent())
+    d = s.dict
+    un = collect(keys(d))
+    rmax = maximum(ndigits(k) for k in values(d))
+    offmax = szw - (rmax + 3)
+    if length(un) <= szh
+      lmax = min(maximum(length(sprint(show, a)) for a in un), offmax)
+      for k in un
+        pk = sprint(show, k)
+        lk = length(pk)
+        v = d[k]
+        println(io)
+        if lk > offmax
+          print(io, pk[1:offmax-length(" \u2026")], " \u2026")
+        else
+          print(io, pk)
+        end
+        lk = min(offmax, lk)
+        if v > 1
+          print(io, " "^(lmax-lk+1), ": $v")
+        end
+      end
+    else
+      lmax = min(maximum(length(sprint(show, a)) for a in un[1:szh]), offmax)
+      for i in 1:szh
+        println(io)
+        k = un[i]
+        pk = sprint(show, k)
+        lk = length(sprint(show, k))
+        v = d[k]
+        if lk > offmax
+          print(io, pk[1:offmax-length(" \u2026")], " \u2026")
+        else
+          print(io, pk)
+        end
+        lk = min(offmax, lk)
+        if v > 1
+          print(io, " "^(lmax-lk+1), ": $v")
+        end
+      end
+      println(io)
+      print(io, "\u22ee")
+    end
+  end
+end
+
+### Iteration
+
 Base.isempty(s::MSet) = isempty(s.dict)
 Base.length(s::MSet) = sum(values(s.dict))
 Base.IteratorSize(::Type{MSet}) = Base.HasLength()
 Base.IteratorEltype(::Type{MSet}) = Base.HasEltype()
 Base.eltype(::Type{MSet{T}}) where {T} = T
-Base.in(x, s::MSet) = haskey(s.dict, x)
+Base.in(x, s::MSet) = any(y -> x == y, keys(s.dict))
```

Comment:
Isn't this much slower than the old version? O(n) Vs O(1)?

Reply:
I haven't tried but the previous version is not practical. See for instance what could happen in a very basic example: 
```
julia> d = Dict{QQFieldElem, Int}(2 => 2)
Dict{QQFieldElem, Int64} with 1 entry:
  2 => 2

julia> haskey(d, 2)
false
```
I have not thought about about a better work-around..